src/Command/SubProfileCommand.php

```diff
@@ -0,0 +1,351 @@
+        $question->setMultiselect(true);
+        $io->writeln('The following components can be excluded: ' . Element::oxford($question->getChoices()));
+        $excluded_dependencies = $io->askChoiceQuestion($question);
+      }
+      $input->setOption('excluded_dependencies', $excluded_dependencies);
+    }
+
+    $excluded_subcomponents = $input->getOption('excluded_subcomponents');
+    if (!$excluded_subcomponents) {
+      if ($io->confirm($this->trans('Would you like to exclude any of Lightning\'s sub-components?'), false)) {
+        // Subcomponents whose parent is already excluded are automatically
+        // excluded. So we build a list of excludable subcomponents whose parent
+        // component isn't already excluded.
+        $excludable_subcomponents = [];
+        $top_level_components = self::getTopLevelComponents();
+        $non_excluded_tlcs = array_diff($top_level_components, $excluded_dependencies);
+        foreach ($non_excluded_tlcs as $non_excluded_tlc) {
+          $tlc_subcomponents = self::getSubComponents($non_excluded_tlc);
+          $excludable_subcomponents = array_merge($excludable_subcomponents, $tlc_subcomponents);
+        }
+
+        $question = new ChoiceQuestion('List the sub-components you would like to exclude, separated by commas.', $excludable_subcomponents);
+        $question->setMultiselect(true);
+        $io->writeln('The following sub-components can be excluded: ' . Element::oxford($question->getChoices()));
+        $excluded_subcomponents = $io->askChoiceQuestion($question);
+      }
+      $input->setOption('excluded_subcomponents', $excluded_subcomponents);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $io = new DrupalStyle($input, $output);
+
+    if (!$this->confirmGeneration($io)) {
+      return;
+    }
+
+    $profile = $this->validator->validateModuleName($input->getOption('profile'));
+    $machine_name = $this->validator->validateMachineName($input->getOption('machine-name'));
+    $description = $input->getOption('description');
+    $profile_path = $this->appRoot . '/profiles/custom';
+
+    // Check if all module dependencies are available.
+    $dependencies = $this->validator->validateModuleDependencies($input->getOption('dependencies'));
+    if ($dependencies) {
+      // @todo ProfileCommand::checkDependencies is private so we aren't
+      // actually checking to see if they are present.
+      $dependencies = $dependencies['success'];
+    }
+    $this->buildExcludedDependenciesList($input->getOption('excluded_dependencies'));
+    $this->buildExcludedDependenciesList($input->getOption('excluded_subcomponents'));
+    $this->generator->generate(
+      $profile,
+      $machine_name,
+      $profile_path,
+      $description,
+      $dependencies,
+      $this->excludedDependencies
+    );
+  }
+
+  /**
+   * @param array $excluded_dependencies
```

Comment:
The method is missing a description, and $excluded_dependencies can probably get a more specific type hint (like string[])